core/controllers/feedback.py

```diff
@@ -57,13 +81,27 @@ class ThreadListHandler(base.BaseHandler):
 
     @acl_decorators.can_play_exploration
     def get(self, exploration_id):
+
+        feedback_thread_dicts = [
+            t.to_dict() for t in feedback_services.get_all_threads(
+            feconf.ENTITY_TYPE_EXPLORATION, exploration_id, False)]
+
+        suggestion_thread_dicts = [
+            t.to_dict() for t in feedback_services.get_all_threads(
+            feconf.ENTITY_TYPE_EXPLORATION, exploration_id, True)]
+
         self.values.update({
-            'feedback_thread_dicts': (
-                [t.to_dict() for t in feedback_services.get_all_threads(
-                    feconf.ENTITY_TYPE_EXPLORATION, exploration_id, False)]),
-            'suggestion_thread_dicts': (
-                [t.to_dict() for t in feedback_services.get_all_threads(
-                    feconf.ENTITY_TYPE_EXPLORATION, exploration_id, True)])
+            'feedback_thread_dicts': get_updated_thread_dict(
+                get_updated_thread_dict(
+                feedback_thread_dicts,
+                'original_author_id',
+                'original_author_username'),
+                'last_nonempty_message_author_id',
+                'last_nonempty_message_author'),
```

Comment:
Can we instead pass tuples into the function and these tuples will be mapping between id and username field?

Reply:
So smth like `get_updated_thread_dict(feedback_thread_dicts, [('a_id', 'a_username'), ('by_id', 'b_username')])`

Reply:
yup, I did. PTAL,